bindings/java/src/test/java/org/apache/opendal/test/behavior/AbstractBehaviorTest.java

```diff
@@ -605,6 +757,158 @@ public void testBlockingCopyOverwrite() {
+                    .is(OpenDALExceptionCondition.ofAsync(OpenDALException.Code.NotFound));
+
+            assertThat(blockingOperator.stat(targetPath).getContentLength()).isEqualTo(sourceContent.length);
+
+            blockingOperator.delete(sourcePath);
+            blockingOperator.delete(targetPath);
+        }
+
+        /**
+         * Rename a nonexistent source should return an error.
+         */
+        @Test
+        public void testBlockingRenameNonExistingSource() {
+            final String sourcePath = UUID.randomUUID().toString();
+            final String targetPath = UUID.randomUUID().toString();
+
+            assertThatThrownBy(() -> blockingOperator.rename(sourcePath, targetPath))
+                    .is(OpenDALExceptionCondition.ofAsync(OpenDALException.Code.NotFound));
+        }
+
+        /**
+         * Rename a dir as source should return an error.
+         */
+        @Test
+        public void testBlockingRenameSourceDir() {
+            final String sourcePath = String.format("%s/", UUID.randomUUID().toString());
+            final String targetPath = UUID.randomUUID().toString();
+
+            blockingOperator.createDir(sourcePath);
+
+            assertThatThrownBy(() -> blockingOperator.rename(sourcePath, targetPath))
+                    .is(OpenDALExceptionCondition.ofAsync(OpenDALException.Code.IsADirectory));
+        }
+
+        /**
+         * Rename to a dir should return an error.
+         */
+        @Test
+        public void testBlockingRenameTargetDir() {
+            final String sourcePath = UUID.randomUUID().toString();
+            final byte[] sourceContent = generateBytes();
+
+            blockingOperator.write(sourcePath, sourceContent);
+
+            final String targetPath = String.format("%s/", UUID.randomUUID().toString());
+
+            blockingOperator.createDir(targetPath);
+
+            assertThatThrownBy(() -> blockingOperator.rename(sourcePath, targetPath))
+                    .is(OpenDALExceptionCondition.ofAsync(OpenDALException.Code.IsADirectory));
+
+            blockingOperator.delete(sourcePath);
+            blockingOperator.delete(targetPath);
+        }
+
+        /**
+         * Rename a file to self should return an error.
+         */
+        @Test
+        public void testBlockingRenameSelf() {
+            final String sourcePath = UUID.randomUUID().toString();
+            final byte[] sourceContent = generateBytes();
+
+            blockingOperator.write(sourcePath, sourceContent);
+
+            assertThatThrownBy(() -> blockingOperator.rename(sourcePath, sourcePath))
+                    .is(OpenDALExceptionCondition.ofAsync(OpenDALException.Code.IsSameFile));
+
+            blockingOperator.delete(sourcePath);
+        }
+
+        /**
+         * Rename to a nested path, parent path should be created successfully.
+         */
+        @Test
+        public void testBlockingRenameNested() {
```

Comment:
The same. Please check opendal's behavior test first before adding bindings tests.